scripts/main/mapview.js

```diff
@@ -264,20 +337,30 @@ mapview.close = function () {
 	$("#mapview").hide();
 	header.setMode("album");
 
-	// Make album focussable
+	// Make album focusable
 	tabindex.makeFocusable(lychee.content);
 };
 
+/**
+ * @param {jQuery} elem
+ * @returns {void}
+ */
 mapview.goto = function (elem) {
 	// If map functionality is disabled -> do nothing
 	if (!lychee.map_display) return;
 
-	var photoID = elem.attr("data-id");
-	var albumID = elem.attr("data-album-id");
+	const photoID = elem.attr("data-id");
+	let albumID = elem.attr("data-album-id");
 
-	if (albumID == "null") albumID = 0;
+	if (albumID === "null") albumID = "unsorted";
 
-	if (album.json == null || albumID !== album.json.id) {
+	// The condition below looks suspicious and like a violation of the
+	// principle of separation of concerns.
+	// In theory, if the currently loaded album does not match the desired
+	// album, then `lychee.goto` and `lychee.load` should take care of that.
+	// But I am afraid of deleting these lines of code and breaking something.
+	// TODO: Clean this up.
+	if (album.json && album.json.id !== albumID) {
 		album.refresh();
 	}
```

Comment:
Your are braver than me, but yes I will do it.

But I wonder if we should that in a separate PR just in case it breaks something unexpectedly. We could "collect" all these lines, which most likely _could_ be deleted, so that is is easier to revert them later.

Reply:
Well, I would say that I've seen some much braver changes in this PR already :smile:.

In principle I prefer not to mix unrelated changes into PRs but this one is so large already that everything will need to be thoroughly retested anyway so I feel that pulling in more at this point will _reduce_ the overall effort.

Reply:
We seem to have different opinions about what means being brave. I tried only the annotate the JS code, but leave it untouched as far as possible and only change what WAS necessary due to the changed API or fix completely obvious bugs.

I don't agree that this fix would count as unrelated. So, we can do it.

Reply:
Done in: https://github.com/LycheeOrg/Lychee-front/pull/282/commits/0afd0a9611ec862597096ad8c8ff4e0356ef6918